src/Extension/CoreExtension.php

```diff
@@ -1300,6 +1302,65 @@ function twig_test_iterable($value)
     return $value instanceof \Traversable || \is_array($value);
 }
 
+/**
+ * Checks if a variable is a sequence.
+ *
+ *    {# evaluates to true if the foo variable is a list or a traversable object #}
+ *    {% if foo is sequence %}
+ *        {# ... #}
+ *    {% endif %}
+ *
+ * @param mixed $value A variable
+ *
+ * @return bool true if the value is a sequence
+ */
+function twig_test_sequence($value)
+{
+    return $value instanceof \Traversable || (\is_array($value) && \array_is_list($value));
```

Comment:
Not all `Traversable` are sequences, they can return any key. Example https://3v4l.org/LJ0R2

The implementation of `array_is_list` could accept any `iterable`, a new name will be required as this is not identical to the PHP 8.1 function.

Reply:
The issue for an iterator is that this requires consuming the iterator, which might not be rewindable to consume it a second time (see generators)